### ISSUE TITLE
import/no-unresolved should work with commonjs by default

### ISSUE THREAD

**Issue by danielo515** [benmosher/eslint-plugin-import#1734]
Hello.
Thank you for this great plugin, really helps me reduce the amount of errors I have.
However, it requires me to configure the no-unresolved rule separatelly to make it work with modules of type commonjs.
I expected that this configuration:

```yaml
extends:
    - plugin:import/errors
    - plugin:import/warning
parserOptions:
    ecmaVersion: 2017
    sourceType: module
    ecmaFeatures:
        spread: true
```

Was all that I may need, but I had to add specific rule configuration like this:

```yaml
rules:
    import/no-unresolved: [2, { commonjs: true }]
```

I think require is common enough to be taken in account by default by just importing the presets. Or at least provide a global setting.

regards

**Comment by ljharb** [comment]
Unfortunately that would be a breaking change, and we're only going to do a breaking change if we are *forced* to.

If there's ever a v3, it will definitely be enabled by default, as will a number of other rule configurations.

**Comment by danielo515** [comment]
I want to understand how can it be a breaking change.
Does it influence on the ES imports?

**Comment by ljharb** [comment]
It'd be a breaking change because CJS code that worked fine would suddenly get warnings on it.

**Comment by danielo515** [comment]
Thanks for clarifying.
Regards
